lumicks/pylake/channel.py

```diff
@@ -796,32 +801,16 @@ def _apply_mask(self, mask):
 
     @staticmethod
     def from_dataset(dset, y_label="y", calibration=None) -> Slice:
-        class LazyLoadedCompoundField:
-            """Wrapper to enable lazy loading of HDF5 compound datasets
-
-            Notes
-            -----
-            We only need to support the methods `__array__()` and `__len__()`, as we only access
-            `LazyLoadedCompoundField` via the properties `TimeSeries.data`, `timestamps` and the
-            method `__len__()`.
-
-            `LazyLoadCompoundField` might be replaced with `dset.fields(fieldname)` if and when the
-            returned `FieldsWrapper` object provides an `__array__()` method itself"""
-
-            def __init__(self, dset, fieldname):
-                self._dset = dset
-                self._fieldname = fieldname
-
-            def __array__(self):
-                """Get the data of the field as an array"""
-                return self._dset[self._fieldname]
-
-            def __len__(self):
-                """Get the length of the underlying dataset"""
-                return len(self._dset)
-
-        data = LazyLoadedCompoundField(dset, "Value")
-        timestamps = LazyLoadedCompoundField(dset, "Timestamp")
+        data = (
+            LazyCache.from_h5py_dset(dset, field="Value")
+            if caching.global_cache
+            else dset.fields("Value")
+        )
+        timestamps = (
+            LazyCache.from_h5py_dset(dset, field="Timestamp")
+            if caching.global_cache
+            else dset.fields("Timestamp")
+        )
```

Comment:
I think we should move the cache checking into functions in the `caching` package.
Those would be concerned about what caching to use while this code is only about loading the channel data.

I added a basic example below, but I think we can add other ones that cover the other usages from loading `numpy` arrays and such.

```py
# in caching
def from_h5py(dset, field=None):
    global global_cache
    return (
        LazyCache.from_h5py_dset(dset, field=field)
        if global_cache
        else dset.fields(field)
    )

# here
    @staticmethod
    def from_dataset(dset, y_label="y", calibration=None) -> Slice:
        data = caching.from_h5py(dset, field="Value")
        timestamps = caching.from_h5py(dset, field="Timestamp")
```

Reply:
Yeah, that's a good point. That keeps it all in one place. Easier to change.

I've added the method you suggested and put the retrieval in `caching` as well. Decided to go with a mixin here, since then we can keep the storage with the code that stores.